sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -139,7 +139,7 @@ namespace Azure { namespace Core {
      * @brief Gets the current #Azure::Core::OperationStatus of the long-running operation.
      *
      */
-    OperationStatus Status() const noexcept { return m_status; }
```

Comment:
This is a fairly significant change because it silently ties the lifetime of the returned `OperationStatus` to the lifetime of the `Operation`. If the `Operation` is deleted, the returned `OperationStatus` is a dangling reference to uninitialized memory.

It's also a breaking change since this is a change to the API signature of a public API.

Reply:
No one should call `operation.Status()` after it has been destructed, or while it is destructing. And no one can expect to have a reference without controlling its lifetime and expect it won't go bad. But in order to do that, you have to be very persistent:
```cpp
OperationStatus s1 = operation.Status(); // creates a copy
auto s2 = operation.Status(); // creates a copy - auto does not declare references even if the type is reference.

OperationStatus const& s3 = operation.Status(); // would break*, but why'd you do that in the past?
decltype(operation.Status()) s4 = operation.Status(); // would break*, but why'd you do decltype when auto is sufficient?

// * Examples where it would break, are only possible if the code does not control the lifetime of an operation and it gets destroyed while the code still has a reference to status.
```

Reply:
Regarding lifetime, 5 operations are statically allocated: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/src/operation_status.cpp#L8-L12

So, in order to get in trouble with the old code after upgrading to new Core version, you should be getting the value as `s3` or `s4` from example above, not control the lifetime of Operation (not have it as copy), and the value should be not the one of the pre-defined 5 operations.

Reply:
You're absolutely right that you need to be persistent. But our customers are VERY good at doing silly things.

What value does making `Status()` return a reference bring to the table, especially since the reference is const (which means that it's value can never change)?

Reply:
It can remain `noexcept`. If we return a copy, constructor may throw: https://en.cppreference.com/w/cpp/string/basic_string/basic_string

So, we should either remove `noexcept` or make it to return const ref.